oauthproxy.go

```diff
@@ -490,6 +493,12 @@ func (p *OAuthProxy) GetRedirect(req *http.Request) (redirect string, err error)
 	}
 
 	redirect = req.Header.Get("X-Auth-Request-Redirect")
+	logger.Printf("X-Auth-Request-Redirect: %s", redirect)
+	// traefik does not send X-Auth-Request-Redirect, so give the traefik headers a try in case X-Auth-Request-Redirect is not set
+	if !p.IsValidRedirect(redirect) {
+		redirect = req.Header.Get("X-Forwarded-Proto") + "://" + req.Header.Get("X-Forwarded-Host") + req.Header.Get("X-Forwarded-Uri")
+		logger.Printf("X-Forwarded-* headers combined: %s", redirect)
+	}
 	if req.Form.Get("rd") != "" {
```

Comment:
Is Traefik not able to explicitly configure the redirect URL in the `rd` param?  For instance, this is what my nginx ingress sign-in annotation looks like:

Internal:
`nginx.ingress.kubernetes.io/auth-signin: https://<REDACTED_DOMAIN>/oauth2/start?rd=https%3A%2F%2F$host$escaped_request_uri`

Behind Proxy w/`X-Forwarded-Host`:
`nginx.ingress.kubernetes.io/auth-signin: https://<REDACTED_DOMAIN>/oauth2/start?rd=https%3A%2F%2F$http_x_forwarded_host$escaped_request_uri`

Reply:
It might with some custom config (just like we are able to do a lot of what this PR adds with custom traefik configs right now), but the whole goal of this PR is to simplify the process of using forward auth. having to manually set explicit url's really defeats that purpose.

Reply:
AFAIK this isn't possible in Traefik's config. (Rather nginx's config is closer to a scripting language so I'm sure we'll be back here once Caddy ships a forward auth middleware.)

There are various feature requests and future plans that might allow this, but there's also a backwards compatibility issue.
Supporting `X-Forwarded-*` would add compatibility for Traefik all the way down to v1.4+ (2017) if I'm not mistaken.

Reply:
> having to manually set explicit url's really defeats that purpose.

Do you mean statically configuring the `rb=` parameter?

That might work for the root of a static setup, but even if the domain and path prefixes would remain static, it wouldn't support different paths. If a client requested `/api/version` and instead ends up at `/` after auth because of the static config, I wouldn't consider fully supported. Especially because this isn't the expected behavior when we're using `X-Auth-Request-Redirect` / `?rb=` in other setups.

Reply:
I was saying the proposed idea was a bad one, so i'm not really into debating how it could be done anyway.

Reply:
Agreed @Ornias1993, sorry for preaching to the choir. I wanted make sure I understood you correctly, and to include some more specifics for anyone else to follow along in case they're less familiar with Traefik.